cmd/root.go

```diff
@@ -18,6 +18,12 @@ var defaultHost = "https://circleci.com"
 // it should be set once when Execute is first called
 var rootCmd *cobra.Command
 
+// config is used internally for preparing CLI and passed to sub-commands
+var config *settings.Config
```

Comment:
In hind-sight `config` is probably not a great name because it creates potential confusion with the name of the argument passed to individual commands.

I'd like to rename this to `rootOptions` in a follow-up PR.

Reply:
https://github.com/CircleCI-Public/circleci-cli/pull/179